### ISSUE TITLE
[4.1] 'Save as Copy' error "Save failed with the following error: The alias ... is already being used by..."

### ISSUE THREAD

**Issue by toivo** [joomla/joomla-cms#37812]
### Steps to reproduce the issue
Edit a menu item for example in the Main Menu.
Click 'Save as Copy' in the drop-down list of 'Save & Close'

### Expected result
A copy of the menu item is saved, similar to Joomla 3.10.9.

### Actual result
Error message "Save failed with the following error: The alias my-menu-item is already being used by My Menu Item menu item in the Main Menu menu (remember it may be a trashed item)".

### System information (as much as possible)
Joomla 4.1.4-dev of 16 May using PHP 8.0.15.

### Additional comments
Reported in the Joomla! Forum topic https://forum.joomla.org/viewtopic.php?f=808&t=993777

**Comment by zero-24** [comment]
Confirmed its an regression of this PR: https://github.com/joomla/joomla-cms/pull/37395 cc @bembelimen 

To me it looks like the original PR fixed the issue with the wrong approach. As described in the PR in the case of save as copy an ID value of "0" will be set and therefor will have an ID set within the data but its "empty" (== 0) and in that case we need the id from the item above.

Before said change we ignored any unsaved changes within com_menus as discribed in the original PR as it was intended to be a 1:1 copy of the existing Item. But we dont do that anymore within [com_content ](https://github.com/joomla/joomla-cms/blob/4.1-dev/administrator/components/com_content/src/Model/ArticleModel.php#L783-L819) there we only update the title and alias in case they already exist an coresponding PR for com_menus will follow
